draft-irtf-cfrg-vdaf.md

```diff
@@ -1040,46 +1040,62 @@ DAFs MUST implement the following function:
-batches are defined by the application), it combines them into a share of the
-desired aggregate result:
+Once an Aggregator holds an output share, it adds it into its aggregate share
+for the batch (where batches are defined by the application).
 
-* `daf.aggregate(agg_param: AggParam, out_shares: list[OutShare]) -> AggShare`
-  is the deterministic aggregation algorithm. It is run by each Aggregator a
-  set of recovered output shares.
+* `daf.agg_init(agg_param: AggParam) -> AggShare` returns an empty aggregate
+  share. It is called to initialize aggregation of a batch of measurements.
 
-~~~~
-    Aggregator 0    Aggregator 1        Aggregator SHARES-1
-    ============    ============        ===================
+* `daf.agg_update(agg_param: AggParam, agg_share: AggShare, out_share:
+  OutShare) -> AggShare` accumulates an output share into an aggregate share
+  and return the updated aggregate share.
 
-    out_share_0_0   out_share_1_0       out_share_[SHARES-1]_0
-    out_share_0_1   out_share_1_1       out_share_[SHARES-1]_1
-    out_share_0_2   out_share_1_2       out_share_[SHARES-1]_2
-         ...             ...                     ...
-    out_share_0_B   out_share_1_B       out_share_[SHARES-1]_B
-      |               |                   |
-      V               V                   V
-    +-----------+   +-----------+       +-----------+
-    | aggregate |   | aggregate |   ... | aggregate |
-    +-----------+   +-----------+       +-----------+
-      |               |                   |
-      V               V                   V
-    agg_share_0     agg_share_1         agg_share_[SHARES-1]
 ~~~~
-{: #aggregate-flow title="Aggregation of output shares. `B` indicates the number of
-measurements in the batch."}
+    Aggregator j
+    ============
+
+    +------------+    +------------+
+    | agg_init   |--->| agg_update |<--- out_share_0
+    +------------+    +------------+
+                         |
+                         V
+                      +------------+
+                      | agg_update |<--- out_share_1,
+                      +------------+
+                        |
+                        V
+                       ...
+                        |
+                        V
+                      +------------+
+                      | agg_update |<--- out_share_MEAS_COUNT
+                      +------------+
+                        |
+                        V
+                      agg_share_j
+~~~~
+{: #aggregate-flow title="Local aggregation of output shares by Aggregator j.
+The number of measurements in the batch is denoted by MEAS_COUNT."}
+
+In many situations it is desirable to split an aggregate share across multiple
+storage elements, then merge the aggregate shares together just before
+releasing the completed aggregate share to the Collector. DAFs facilitate this
+with the following method:
+
+* `daf.merge(agg_param: AggParam, agg_shares: list[AggShare]) -> AggShare`
```

Comment:
nitpicky Q. `agg_update` is specified as a binary operation on an `AggShare` and an `OutShare`; `merge` is specified as acting on a list of `AggShare`s (both operations parameterized on an `AggParam`).

Is there a reason for this difference, i.e. do we think we might be able to implement more efficient merging than if we specified `merge(agg_param, agg_share_left, agg_share_right)`?

Reply:
Motivation: I think this will be helpful for DAP, where we will have a sequence of aggregate shares, one for each "bucket", where a collection request specifies a sequence of buckets (e.g., a a time interval specifies a sequence of non-overlapping time windows).

Another benefit is that we can use `merge()` in the implementation of `unshard()` for both Prio3 and Poplar1.

Is there any benefit to making`merge()` a binary operator besides symmetry?

Reply:
Nope, symmetry was my only concern, and making sure I wasn't missing anything. I don't think any of those uses would be hindered by making this a binary operator (I suppose it's a matter of where the `for` loop goes), but I also don't think this is important enough to hold this PR up over.